packages/ui/src/components/va-virtual-scroller/VaVirtualScroller.vue

```diff
@@ -0,0 +1,145 @@
+    }
+    if (!props.disabled) { useEvent('scroll', handleScroll, true) }
+
+    const { list, wrapper, itemSize, wrapperSize } = useVirtualScrollerSizes(props, listScrollPosition)
+
+    const uniqueKey = (item: unknown, index: number) => {
+      if (props.trackBy && item && typeof item === 'object') {
+        const isArrayItem = Array.isArray(item)
+
+        let key: any
+        if (isArrayItem && !isNaN(+props.trackBy)) { key = (item as any[])[+props.trackBy] }
+        if (!isArrayItem) { key = (item as Record<string, any>)[props.trackBy] }
+
+        if (key || key === 0) { return key }
+
+        warn(`[va-virtual-scroller] ${isArrayItem ? 'Index' : 'Key'} '${props.trackBy}' wasn't found in provided ${isArrayItem ? 'array' : 'object'}: `, item)
+      }
+
+      return renderStartIndex.value + index
+    }
+
+    // forming items to render
+    const renderStartIndex = computed(() => {
+      return Math.max(0, Math.floor(listScrollPosition.value / itemSize.value) - props.bench)
+    })
+    const renderItemsAmount = computed(() => {
+      if (!props.items?.length) { return 0 }
+      return props.disabled
+        ? props.items.length
+        : Math.min(props.items.length - renderStartIndex.value, Math.ceil(wrapperSize.value / itemSize.value) + props.bench * 2)
+    })
+    const renderEndIndex = computed(() => renderStartIndex.value + renderItemsAmount.value)
+    const renderBuffer = computed(() => {
+      if (!props.items?.length) { return [] }
+      return props.items.slice(renderStartIndex.value, renderEndIndex.value)
+    })
+
+    const sizeAttribute = computed(() => props.horizontal ? 'width' : 'height')
+
+    // wrapper styles
+    const wrapperStyleComputed = computed(() => ({
+      [sizeAttribute.value]: `${wrapperSize.value}px`,
+    }))
+
+    // container styles
+    const containerSize = computed(() => (props.items?.length ?? 0) * itemSize.value)
+    const containerStyleComputed = computed(() => ({
+      [sizeAttribute.value]: `${containerSize.value}px`,
+    }))
+
+    // items list classes and styles
+    const currentListOffset = computed(() => renderStartIndex.value * itemSize.value)
+    const listStyleComputed = computed(() => ({
+      transform: `translate${props.horizontal ? 'X' : 'Y'}(${currentListOffset.value}px)`,
+    }))
+    const listClassComputed = useBem('va-virtual-scroller', () => ({
+      ...pick(props, ['horizontal']),
+    }))
+
+    return {
+      containerStyleComputed,
+      wrapperStyleComputed,
+      listClassComputed,
+      listStyleComputed,
+      renderStartIndex,
+      renderBuffer,
+      uniqueKey,
+      wrapper,
+      list,
+    }
+  },
+})
+</script>
+
+<style lang="scss">
+  .va-virtual-scroller {
+    overflow: auto;
+
+    &__container {
+      position: relative;
+      overflow: hidden;
+    }
+
+    &__list {
+      display: flex;
+      flex-direction: column;
+    }
+
+    &--horizontal {
+      &.va-virtual-scroller__list {
```

Comment:
Isn't it will be `va-virtual-scroller--horizontal va-virtual-scroller__list`?

Reply:
And what's the problem? BEM doesn't welcome structures with multiply elements or modifiers.

Reply:
Yes it's worked because i have forgotten to move root modifier class to the root div :)